src/main/java/net/dv8tion/jda/core/managers/GuildController.java

```diff
@@ -2229,6 +2230,71 @@ protected void handleResponse(Response response, Request<Emote> request)
         return new ChannelOrderAction<>(guild, ChannelType.VOICE);
     }
 
+    /**
+     * Modifies the positional order of {@link net.dv8tion.jda.core.entities.Category#getTextChannels() Category#getTextChannels()}
+     * using an extension of {@link net.dv8tion.jda.core.requests.restaction.order.ChannelOrderAction ChannelOrderAction}
+     * specialized for ordering the nested {@link net.dv8tion.jda.core.entities.TextChannel TextChannels} of this
+     * {@link net.dv8tion.jda.core.entities.Category Category}.
+     * <br>Like {@code ChannelOrderAction}, the returned {@link net.dv8tion.jda.core.requests.restaction.order.CategoryOrderAction CategoryOrderAction}
+     * can be used to move TextChannels {@link net.dv8tion.jda.core.requests.restaction.order.OrderAction#moveUp(int) up},
+     * {@link net.dv8tion.jda.core.requests.restaction.order.OrderAction#moveDown(int) down}, or
+     * {@link net.dv8tion.jda.core.requests.restaction.order.OrderAction#moveTo(int) to} a specific position.
+     * <br>This uses <b>ascending</b> order with a 0 based index.
+     *
+     * <p>Possible {@link net.dv8tion.jda.core.requests.ErrorResponse ErrorResponses} include:
+     * <ul>
+     *     <li>{@link net.dv8tion.jda.core.requests.ErrorResponse#UNKNOWN_CHANNEL UNNKOWN_CHANNEL}
+     *     <br>One of the channels has been deleted before the completion of the task.</li>
+     *
+     *     <li>{@link net.dv8tion.jda.core.requests.ErrorResponse#MISSING_ACCESS MISSING_ACCESS}
+     *     <br>The currently logged in account was removed from the Guild.</li>
+     * </ul>
+     *
+     * @param  category
+     *         The {@link net.dv8tion.jda.core.entities.Category Category} to order
+     *         {@link net.dv8tion.jda.core.entities.TextChannel TextChannels} from.
+     *
+     * @return {@link net.dv8tion.jda.core.requests.restaction.order.CategoryOrderAction CategoryOrderAction} - Type: {@link net.dv8tion.jda.core.entities.TextChannel TextChannel}
+     */
+    @CheckReturnValue
+    public CategoryOrderAction<TextChannel> modifyTextChannelPositions(Category category)
+    {
+        return new CategoryOrderAction<>(category, ChannelType.TEXT);
```

Comment:
Check that the provided Category is from the Guild that this GuildController is associated with.